mediapipe/graphs/instantmotiontracking/calculators/matrices_manager_calculator.cc

```diff
@@ -0,0 +1,324 @@
+  return ::mediapipe::OkStatus();
+}
+
+::mediapipe::Status MatricesManagerCalculator::Process(CalculatorContext* cc) {
+  auto model_matrices = std::make_unique<TimedModelMatrixProtoList>();
+  TimedModelMatrixProtoList* model_matrix_list = model_matrices.get();
+  model_matrix_list->clear_model_matrix();
+
+  const std::vector<UserRotation> user_rotation_data =
+      cc->Inputs().Tag(kUserRotationsTag).Get<std::vector<UserRotation>>();
+
+  const std::vector<UserScaling> user_scaling_data =
+        cc->Inputs().Tag(kUserScalingsTag).Get<std::vector<UserScaling>>();
+
+  const std::vector<int> render_data =
+      cc->Inputs().Tag(kRendersTag).Get<std::vector<int>>();
+
+  const std::vector<Anchor> anchor_data =
+      cc->Inputs()
+          .Tag(kAnchorsTag)
+          .Get<std::vector<Anchor>>();
+
+  // Device IMU Data definitions
+  const float yaw = cc->Inputs().Tag(kIMUDataTag).Get<float[]>()[0];
+  const float pitch = cc->Inputs().Tag(kIMUDataTag).Get<float[]>()[1];
+  const float roll = cc->Inputs().Tag(kIMUDataTag).Get<float[]>()[2];
+  // We can pre-generate the IMU submatrix as this remains unchanged for all objects
+  const Matrix3f imu_rotation_submatrix = GenerateIMURotationSubmatrix(yaw, pitch, roll);
+
+  int render_idx = 0;
+  for (const Anchor &anchor : anchor_data) {
+    const int id = anchor.sticker_id;
+
+    TimedModelMatrixProto* model_matrix =
+        model_matrix_list->add_model_matrix();
+    model_matrix->set_id(id);
+
+    // The user transformation data associate with this sticker must be defined
+    const float rotation = GetUserRotation(user_rotation_data, id);
+    const float scaler = GetUserScaler(user_scaling_data, id);
+
+    // A vector representative of a user's sticker rotation transformation can be created
+    const Matrix3f user_rotation_submatrix = GenerateUserRotationMatrix(rotation);
+    // Next, the submatrix representative of the user's scaling transformation must be generated
+    const DiagonalMatrix3f user_scaling_submatrix = GenerateScalingMatrix(scaler);
+
+    // The user transformation data can be concatenated into a final rotation submatrix with the
+    // device IMU rotational data
+    const Matrix3f user_transformed_rotation_submatrix = user_scaling_submatrix * imu_rotation_submatrix * user_rotation_submatrix;
+
+    // We can also generate the initial scaling diagonal based on the preset scale factor for this render id
+    const DiagonalMatrix3f render_scaling_diagonal = GenerateScalingMatrix(GetDefaultRenderScale(render_data[render_idx++]));
+
+    // A vector representative of the translation of the object in OpenGL coordinate space must be generated
+    const Vector3f translation_vector = GenerateAnchorVector(anchor);
+
+    // Concatenate all model matrix data
+    const Matrix4fCM final_model_matrix = GenerateEigenModelMatrix(translation_vector, render_scaling_diagonal * user_transformed_rotation_submatrix);
```

Comment:
Can you move user_scaling_matrix and render_scaling_diagonal to the right?
imu_rotation_submatrix * user_rotation_matrix * user_scaling_submatrix * render_scaling_diagonal

The results are the same. But it probably reads better. The model is multiplied on the right, so it reads like the model is first magnified according to some prefixed value, then magnified by the user, then rotated by the user, then adapts to device rotation.